ocpp/v201/datatypes.py

```diff
@@ -0,0 +1,809 @@
+    GetVariablesResponse.GetVariableResultType,
+    NotifyMonitoringReportRequest.MonitoringDataType,
+    NotifyReportRequest.ReportDataType,
+    SetVariableMonitoringRequest.SetMonitoringDataType,
+    SetVariableMonitoringResponse.SetMonitoringResultType,
+    SetVariablesRequest.SetVariableDataType,
+    SetVariablesResponse.SetVariableResultType,
+    NotifyEventRequest.EventDataType
+    """
+
+    name: str
+    instance: Optional[str] = None
+    evse: Optional[EVSEType] = None
+
+
+@dataclass
+class VariableType:
+    """
+    Reference key to a component-variable.
+    VariableType is used by: ComponentVariableType,
+    GetVariablesRequest.GetVariableDataType,
+    GetVariablesResponse.GetVariableResultType,
+    NotifyMonitoringReportRequest.MonitoringDataType,
+    NotifyReportRequest.ReportDataType,
+    SetVariableMonitoringRequest.SetMonitoringDataType,
+    SetVariableMonitoringResponse.SetMonitoringResultType,
+    SetVariablesRequest.SetVariableDataType,
+    SetVariablesResponse.SetVariableResultType,
+    NotifyEventRequest.EventDataType
+    """
+
+    name: str
+    instance: Optional[str] = None
+
+
+@dataclass
+class ComponentVariableType:
+    """
+    Class to report components, variables and variable attributes and
+    characteristics.
+    ComponentVariableType is used by: GetMonitoringReportRequest,
+    GetReportRequest
+    """
+
+    component: ComponentType
+    variable: Optional[VariableType] = None
+
+
+@dataclass
+class CompositeScheduleType:
+    """
+    CompositeScheduleType is used by: GetCompositeScheduleResponse
+    """
+
+    evse_id: int
+    duration: int
+    schedule_start: datetime
+    charging_rate_unit: enums.ChargingRateUnitType
+    charging_schedule_period: ChargingSchedulePeriodType
+
+
+@dataclass
+class CostType:
+    """
+    CostType is used by: ConsumptionCostType
+    """
+
+    cost_kind: enums.CostKindType
+    amount: int
+    amount_multiplier: Optional[int] = None
+
+
+@dataclass
+class ConsumptionCostType:
+    """
+    ConsumptionCostType is used by: SalesTariffEntryType
+    """
+
+    start_value: float
+    cost: CostType
+
+
+@dataclass
+class EventDataType:
+    """
+    Class to report an event notification for a component-variable.
+    EventDataType is used by: NotifyEventRequest
+    """
+
+    event_id: int
+    timestamp: datetime
+    trigger: enums.EventTriggerType
+    actual_value: str
+    event_notification_type: enums.EventNotificationType
+    component: ComponentType
+    variable: VariableType
+    cause: Optional[int] = None
+    tech_code: Optional[str] = None
+    cleared: Optional[bool] = None
+    transaction_id: Optional[str] = None
+    variable_monitoring_id: Optional[int] = None
+
+
+@dataclass
+class FirmwareType:
+    """
+    Represents a copy of the firmware that can be loaded/updated on the
+    Charging Station.
+    FirmwareType is used by: UpdateFirmwareRequest
+    """
+
+    location: str
+    retrival_date_time: datetime
```

Comment:
From the document I have its listed as `retrieveDateTime`. Searching the document there is no `retrieveDataTime`.

Reply:
I see that I introduced a typo in my comment. I meant to write `retrieveDateTime` instead of `retrieveDataTime`. I wanted to emphasize that `retrival` should be changed to `retrieve`. 

![image](https://user-images.githubusercontent.com/1565144/132647004-28544d5b-f310-4cd8-bd4b-23600603e696.png)